langchain4j-core/src/main/java/dev/langchain4j/agent/tool/ToolSpecifications.java

```diff
@@ -44,6 +67,23 @@ public static List<ToolSpecification> toolSpecificationsFrom(Object objectWithTo
         return toolSpecificationsFrom(objectWithTools.getClass());
     }
 
+    /**
+     * Validates all the {@link ToolSpecification}s. The validation checks for duplicate method names.
+     * Throws {@link IllegalArgumentException} if validation fails
+     *
+     * @param toolSpecifications list of ToolSpecification to be validated.
+     */
+    public static void validateSpecifications(List<ToolSpecification> toolSpecifications) throws IllegalArgumentException {
```

Comment:
This method is not used by `AiServices`, `toolSpecificationFrom(Method)` is.
I guess we will need to validate it there...

Reply:
Yes, I had a look at it, but will double check with you where to do the validation. But that will be another PR